core/domain/feedback_services.py

```diff
@@ -366,6 +366,34 @@ def _get_thread_from_model(thread_model):
         thread_model.created_on, thread_model.last_updated)
 
 
+def get_messages_for_exp_id(exp_id, limit=None):
```

Comment:
This is probably OK -- it doesn't fully fix the issue, but you at least limit to 3 messages (say) per thread. That said, would it make more sense to show at most one message per open thread -- just the last one? If so you could just get the N most recently-updated threads and then grab their last messages.